### PR TITLE
Set the C++ standard to C++0x everywhere

### PR DESCRIPTION
**Pull request by rwy0717**
Signed-off-by: Robert Young <rwy0717@gmail.com>

### CODE REVIEW

**Comment by charliegracie** [comment]
@genie-omr build all

File: cmake/modules/platform/toolcfg/gnu.cmake [hunk 1]
```diff
@@ -68,7 +70,7 @@ list(APPEND TR_COMPILE_OPTIONS
 # TR_CXX_COMPILE_OPTIONS are appended to CMAKE_CXX_FLAGS, and so apply only to
 # C++ file compilation
 list(APPEND TR_CXX_COMPILE_OPTIONS
-	-std=c++0x
+	# nothing
```
Comment (charliegracie) [inline cmake/modules/platform/toolcfg/gnu.cmake]: Can you remove the TR_CXX_COMPILE_OPTIONS?

**Comment by charliegracie** [comment]
@genie-omr build all

File: cmake/modules/platform/toolcfg/gnu.cmake (hunk 1 above)
Comment (rwy0717) [reply cmake/modules/platform/toolcfg/gnu.cmake]: done

**Comment by rwy0717** [comment]
Somehow I managed to add a backtick to the end of aix.cmake, fixed now. The linux builds failed to clone, seems there's an infrastructure issue? While I was at it, I rebased as well.

This is ready for a build.

**Pull request merged**